src/App.js

```diff
@@ -1,25 +1,50 @@
-import logo from './logo.svg';
+/* eslint-disable no-unused-vars */
+import React, { useState } from 'react';
+import foods from "./foods.json";
 import './App.css';
+import FoodBox from './Components/FoodBox';
+import AddFoodForm from './Components/AddForm';
+import Search from './Components/Search';
 
 function App() {
+  const [foodList, setFoodList] = useState(foods);
+  const [filteredFoodList, setFilteredFoodList] = useState(foods);
+
+  const handleDelete = (index) => {
+    const updatedFoodList = [...foodList];
+    updatedFoodList.splice(index, 1);
+    setFoodList(updatedFoodList);
+    setFilteredFoodList(updatedFoodList);
+  };
+
+  const handleAddFood = (newFood) => {
+    setFoodList([...foodList, newFood]);
+    setFilteredFoodList([...foodList, newFood]);
+  };
+
+  const handleSearch = (searchTerm) => {
+    const filteredFoods = foodList.filter((food) =>
+      food.name.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+    setFilteredFoodList(filteredFoods);
+  };
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>Food List</h1>
+      <Search handleSearch={handleSearch} />
```

Comment:
Here you're passing props with the name `handleSearch` but inside the component you calling it `onSearch` change the name of the props here and your search bar will work:
```
<Search onSearch={handleSearch} />
```